src/NServiceBus.Transport.Msmq/DeadLetterQueueOptionExtensions.cs

```diff
@@ -18,8 +19,17 @@ public static class DeadLetterQueueOptionExtensions
         public static void UseDeadLetterQueue(this ExtendableOptions options, bool enable = true)
         {
             Guard.AgainstNull(nameof(options), options);
-            var ext = options.GetExtensions();
-            ext.Set(KeyDeadLetterQueue, enable);
+            options.GetDispatchProperties()[KeyDeadLetterQueue] = enable.ToString();
```

Comment:
Nitpicking: do we need to pass invariant culture here, just in case?

Reply:
Good point but I don't think we do, bool's `ToString` implementation is basically hardcoded: https://referencesource.microsoft.com/#mscorlib/system/boolean.cs,6c8fb8c92a24ec37